src/sentry/auth/access.py

```diff
@@ -558,26 +554,43 @@ def has_project_access(self, project: Project) -> bool:
         return project.id in self.project_ids_with_team_membership
 
     def has_any_project_scope(self, project: Project, scopes: Collection[str]) -> bool:
+        """
+        Represent if a user should have access with any one of the given scopes to
+        information for the given project.
+
+        For performance's sake, prefer this over multiple calls to `has_project_scope`.
+        """
         if not self.has_project_access(project):
             return False
-
         if any(self.has_scope(scope) for scope in scopes):
             return True
 
         if self.rpc_user_organization_context.member and features.has(
             "organizations:team-roles", self.rpc_user_organization_context.organization
         ):
-            memberships = self.rpc_user_organization_context.member.member_teams
-            for membership in memberships:
-                team_scopes = membership.scopes
+            with sentry_sdk.start_span(op="check_access_for_all_project_teams") as span:
+                project_teams_id = [p.id for p in project.teams.all()]
```

Comment:
This query would be cheaper as `.values_list("id", flat=True)`.

Also, nit: since we're doing repeated `in` operations below, a set is a little better than a list.